From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Fri, 29 Dec 2023 13:03:42 +0100
Subject: [PATCH] Hide irrelevant compilation warnings

Hides some compilation warnings.
This is done because Paper code contains many instances that trigger these warnings, leading to them filling the terminal every time it is compiled, and making it harder to find errors among the wall of text.

This patch has an analogous patch in `fiddle-server` with the same name.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Hide irrelevant compilation warnings"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index 149f9088fe806467656e8b1c4157df60fda69ba7..41c9fbdb1e86deb47545659fc7222056f616bea7 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -89,6 +89,15 @@ val generateApiVersioningFile by tasks.registering {
     }
 }
 
+// Fiddle start - Gale - hide irrelevant compilation warnings
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("-Xlint:-module")
+    compilerArgs.add("-Xlint:-removal")
+    compilerArgs.add("-Xlint:-dep-ann")
+}
+// Fiddle end - Gale - hide irrelevant compilation warnings
+
 tasks.jar {
     from(generateApiVersioningFile.map { it.outputs.files.singleFile }) {
         into("META-INF/maven/${project.group}/${project.name}")
